forward43/api_ulule.py

```diff
@@ -33,16 +55,37 @@ def main():
 if __name__ == '__main__':
     # main()
     url = 'https://api.ulule.com/v1/'
-    projects = 'search/projects/'
-    query = 'status=all'
+    endpoint_projects = 'search/projects/'
+    query = ['status:currently']
 
-    response = requests.get(url + projects, params={'q': query})
-    if response.status_code != 200:
-        raise Exception('GET {}'.format(response.status_code))
-    # PAY ATTENTION: The response is paginated
+    response = get_page(url + endpoint_projects, query, limit=1, page=0)
 
     data = response.json()
-
-    # TODO: keep only criteria fields
-    # TODO: save to file
+    meta = data['meta']
+    n_projects = meta['total_count']
+    limit = meta['limit']
+    n_pages = math.ceil(n_projects / limit)
+    
+    # %%   Get list of IDs
+    id_list = []
+    for n in range(n_pages):
+        page = get_page(url + endpoint_projects, query, limit, page=n).json()
+        projects = page['projects']
+        for p in projects:
+            id_list.append(p['id'])
+    
+    # %%   Get projects
+    project_list = []
+    for project_id in id_list[:5]:
+        project = get_object(url + f'projects/{project_id}').json()
+        project = {k: project[k] for c in criteria for k in project.keys() 
+                   if k.startswith(c) and len(k) < 15}     # For now: keep all description and name fields
+        project['location'] = project['location']['city']
```

Comment:
This might break if there's no city in the location. If you're sure it always has a city, keep it, otherwise:
`project["location"] = project["location"].get("city", "no location given")`

Reply:
Implemented suggestion (with fixup! 😄), see `ac55ffa`